### DISCUSSION TITLE
Nested relationships

### DISCUSSION THREAD

**Discussion by Dogacel** [openfga/community#326] [Ideas]
### Checklist

- [X] I agree to the terms within the [OpenFGA Code of Conduct](https://github.com/openfga/.github/blob/main/CODE_OF_CONDUCT.md).

### Describe the problem you'd like to have solved

There should be a way to inherit certain properties from relations. This makes sense for higher level entities such as namespaces, organization and other wrapper types in the hierarchy.

```
`view` is an impossible relation for `file` (no entrypoint).ModelValidationError
```

### Describe the ideal solution

```openfga-dsl
model
  schema 1.1

type user

type organization
  relations
    define member: [user]

type folder
  relations
    define organization: [organization]

type file
  relations
    define parent: [folder]
    define organization: organization from parent
    define view: member from organization
    # or
    define view: member from parent from organization
```

### Alternatives and current workarounds

Currently it is possible to push the permission check to the parent,

```openfga-dsl
model
  schema 1.1

type user

type organization
  relations
    define member: [user]

type folder
  relations
    define organization: [organization]
    define is_organization_member: member from organization

type file
  relations
    define parent: [folder]
    define view: is_organization_member from parent 
```

However this gets dirty too quickly, especially when hierarchy is deeper and more permissions are added. Each entity carries over a piece of information that it usually doesn't care.

### Additional context

I would be glad if you can provide some alternatives to this problem, I have also considered other ReBAC solutions and couldn't find a solution that really works with nested relations. Maybe it is related to cache performance, however being able to define nested relations on leaf entities should make caching easier.

Easiest way to implenet this would be expanding the syntax I provided `target: x from y from z` onto
```openfga-dsl
type y
  relations
    define x: [x]
type z
  relations
    define y: [y]
    define x_from_y: x from y
type child
  relations
    define z: [z]
    define target: x_from_y from z
```

Also LMK if there are any performance concerns about this or the previous approach.

**Comment by aaguiarz** [comment]
Thanks for the feedback @Dogacel! I'm moving this to an "Idea"

**Comment by aaguiarz** [comment]
BTW, the approach you are following is correct.
